### PR TITLE
ENH: use attribution from provider if available

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Work in progress, still needs to add tests and update docs. And was trying out if I could "auto wrap" the text, but not yet much success (https://matplotlib.org/3.1.1/gallery/text_labels_and_annotations/autowrap.html)

### CODE REVIEW

**Comment by darribas** [comment]
This looks good to me. I'd only say maybe, if we bring in #94 it should be included here too?

**Comment by jorisvandenbossche** [comment]
>  I'd only say maybe, if we bring in #94 it should be included here too?

Not fully sure what you mean? We will merge both PRs, and depending on which is merged first, the other will need to fix the conflicts?

File: contextily/plotting.py [hunk 1]
```diff
@@ -146,6 +148,11 @@ def add_basemap(
     if reset_extent:
         ax.axis((xmin, xmax, ymin, ymax))
 
+    # Add attribution text
+    if url is None:
+        url = providers.Stamen.Terrain
+    if isinstance(url, dict) and attribution is None:
+        attribution = url.get("attribution")
```
Comment (jorisvandenbossche) [inline contextily/plotting.py]: It's a little bit unfortunate that we need to do the above in more than one place, but I don't think this is avoidable

File: contextily/place.py [hunk 2]
```diff
@@ -178,13 +185,7 @@ def __repr__(self):
...
 
 def plot_map(
-    place,
-    bbox=None,
-    title=None,
-    ax=None,
-    axis_off=True,
-    latlon=True,
-    attribution=ATTRIBUTION,
+    place, bbox=None, title=None, ax=None, axis_off=True, latlon=True, attribution=None
```
Comment (darribas) [inline contextily/place.py]: This is not `black` compatible?

File: contextily/plotting.py (hunk 1 above)
Comment (darribas) [reply contextily/plotting.py]: Could we maybe turn it into a method that we can use throughout? Something like:

```python
def check_attribution(url, attribution):
    if url is None:
        url = providers.Stamen.Terrain
    if isinstance(url, dict) and attribution is None:
        attribution = url.get("attribution")
    return url, attribution
```

File: contextily/plotting.py [hunk 3]
```diff
@@ -157,39 +164,40 @@ def _reproj_bb(left, right, bottom, top, s_crs, t_crs):
...
 
     Arguments
     ---------
     ax                  : AxesSubplot
                           Matplotlib axis with `x_lim` and `y_lim` set in Web
                           Mercator (EPSG=3857)
-    att                 : str
-                          [Optional. Defaults to standard `ATTRIBUTION`] Text to be added at the
-                          bottom of the axis.
+    text                : str
```
Comment (darribas) [inline contextily/plotting.py]: I like this renaming, makes more sense

Comment (darribas) [reply contextily/plotting.py]: done

File: contextily/plotting.py (hunk 1 above)
Comment (jorisvandenbossche) [reply contextily/plotting.py]: For now it is only in 2 places. So maybe let's factor out once there is a third ;)

File: contextily/place.py (hunk 2 above)
Comment (jorisvandenbossche) [reply contextily/place.py]: It's black that did it, so I suppose yes :)

(I think the rule is: if it fits on one line, use one line, if not, put each keyword on its own line)

**Comment by jorisvandenbossche** [comment]
OK, given the minor comments are somewhat answered, will merge this now

**Pull request merged**